File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.cpp
/* A binary tree node

struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};
 */

class Solution {
  public:
    void lv(Node* root,int level,vector<int> &ans){
        if(root==NULL) return ;
        if(level==ans.size()){
            ans.push_back(root->data);
        }
        lv(root->left,level+1,ans);
        lv(root->right,level+1,ans);
    }
    
    vector<int> leftView(Node *root) {
        // code here
        vector<int> ans;
        lv(root,0,ans);
        return ans;
    }
};

File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.py
'''
# Node Class:
class Node:
    def _init_(self,val):
        self.data = val
        self.left = None
        self.right = None
'''
from collections import deque
class Solution:
    def leftview(self,root,level,res):
        if root is None:
            return res
        
        if len(res)==level:
            res.append(root.data)
        
        self.leftview(root.left,level+1,res)
        self.leftview(root.right,level+1,res)
        
        
    def LeftView(self, root):
        # code here
        res=[]
        self.leftview(root,0,res)
        return res
        
            
        